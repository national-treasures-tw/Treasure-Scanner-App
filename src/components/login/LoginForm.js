// @flow
import React from 'react'
import {
  View,
  Text,
  Image,
  TextInput,
  TouchableOpacity
} from 'react-native';

import {
  AuthenticationDetails,
  CognitoUser,
  CognitoUserPool,
  CognitoUserAttribute
} from 'react-native-aws-cognito-js';
import { Images } from '../Themes'
import styles from './styles/LoginFormStyle';
import { onSignIn } from "../auth";

const userPool = new CognitoUserPool({
  UserPoolId: 'us-east-1_8JaJl8ZVD',
  ClientId: '428sfq1asso7a3pam8ugmmssdh',
});

class LoginForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }

  onEmailChange = (email) => {
    this.setState({ email });
  }

  onPasswordChange = (password) => {
    this.setState({ password });
  }

  onLogin = () => {
    const { email, password } = this.state;
    const authenticationData = {
      Username: email,
      Password: password,
    };
    const authenticationDetails = new AuthenticationDetails(authenticationData);

    const userData = {
      Username: email,
      Pool: userPool
    };
    const cognitoUser = new CognitoUser(userData);
    cognitoUser.authenticateUser(authenticationDetails, {
      onSuccess: (result) => {
        console.log(result);
        const token = result.getIdToken().getJwtToken();
        console.log('ID token + ' + token);
        this.props.signIn(token);
        onSignIn(token).then(() => this.props.navigation.navigate("SignedIn"));
      },
      onFailure: (err) => {
        alert(err);
      },
    });
  }
  render () {
    const {
      fetching,
      email,
      password,
      disabled,
      handleChangeEmail,
      handleChangePassword,
      emailError,
      passwordError,
      // submitError,
      onLoginButtonPressed,
      onCancelButtonPressed,
      onSignupButtonPressed,
      onForgotPassword
    } = this.props
    const editable = !fetching
    const textInputStyle = editable ? styles.textInput : styles.textInputReadonly
    const showEmailError = emailError && emailError.message.length > 0
    const showPasswordError = passwordError && passwordError.message.length > 0
    return (
      <View style={styles.form}>

        <View style={[styles.row, showEmailError ? styles.redline : styles.greyline]}>
          <TextInput
            ref='email'
            style={textInputStyle}
            value={email}
            editable={editable}
            keyboardType='email-address'
            returnKeyType='next'
            autoCapitalize='none'
            autoCorrect={false}
            onChangeText={this.onEmailChange}
            underlineColorAndroid='transparent'
            onSubmitEditing={() => this.refs.password.focus()}
            placeholder={'信箱'} />
        </View>

        <View style={[styles.row, showPasswordError ? styles.redline : styles.greyline]}>
          <TextInput
            ref='password'
            style={textInputStyle}
            value={password}
            editable={editable}
            keyboardType='default'
            returnKeyType='go'
            autoCapitalize='none'
            autoCorrect={false}
            secureTextEntry
            onChangeText={this.onPasswordChange}
            underlineColorAndroid='transparent'
            onSubmitEditing={onLoginButtonPressed}
            placeholder={'密碼'} />
        </View>

        <View style={styles.loginColumn}>
          <TouchableOpacity style={styles.loginButton} onPress={this.onLogin}>
            <Text style={styles.loginText}>{'登入'}</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.socialLoginColumn}>
          <TouchableOpacity style={styles.googleButton} onPress={(fetching || disabled) ? () => {} : onLoginButtonPressed}>
            <Text style={styles.loginTextSocial}>{'GOOGLE'}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.facebookButton} onPress={(fetching || disabled) ? () => {} : onLoginButtonPressed}>
            <Text style={styles.loginTextSocial}>{'FACEBOOK'}</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.signupPrompt}>
          <TouchableOpacity style={styles.signupButton} onPress={this.props.onSignupButtonPressed}>
            <Text style={styles.loginTextSocial}>{'註冊'}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.forgotButton} onPress={(fetching || disabled) ? () => {} : onLoginButtonPressed}>
            <Text style={styles.loginTextSocial}>{'忘記密碼'}</Text>
          </TouchableOpacity>
        </View>

      </View>
    )
  }
}

export default LoginForm
