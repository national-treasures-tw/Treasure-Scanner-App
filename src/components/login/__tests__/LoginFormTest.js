import React from 'react'
import LoginForm from '../LoginForm'
import renderer from 'react-test-renderer'

let wrapper = 0

// let s;

beforeEach(() => {
  const loginProps = {
    email: 'Daredevil',
    fetching: true,
    password: "The devil of Hell's Kitchen"
  }
  wrapper = renderer.create(<LoginForm {...loginProps} />).toJSON()
})

test('component exists', () => {
  expect(wrapper).toMatchSnapshot()
})

// test('text inputs are not editable when fetching', () => {
//   const inputs = wrapper.find('View > View > TextInput')
//   inputs.forEach(w => expect(w.props().editable).toBe(false))
// })

// test('email inputs placeholder is correct', () => {
//   const email = wrapper.find('View > View > TextInput').first()
//   expect(email.props().placeholder).toBe(I18n.t('email').toLowerCase())
// })

// test('password inputs placeholder is correct', () => {
//   const password = wrapper.find('View > View > TextInput').last()
//   expect(password.props().placeholder).toBe(I18n.t('password').toLowerCase())
// })

// test('has a login button', () => {
//   const button = wrapper.find('View > View > TouchableOpacity > Text').first()
//   expect(button.props().children).toBe(I18n.t('login').toUpperCase())
// })

// test('email not editable when fetching', () => {
//   wrapper.setProps({ fetching: true })
//   const email = wrapper.find('View > View > TextInput').first()
//   expect(email.props().editable).toBe(false)
// })

// test('password is not editable when fetching', () => {
//   wrapper.setProps({ fetching: true })
//   const password = wrapper.find('View > View > TextInput').last()
//   expect(password.props().editable).toBe(false)
// })

// test('use can forgot password', t => {
//   const button = wrapper.find('View > View > TouchableOpacity').first()
//   button.props().onPress()
//   button.simulate('press')
//   t.is(x, 1)
// })

/**
 * FIXME: test text input
 */

// test('login button can be disabled', t => {
//   // const onLoginButtonPressed = () => {console.error('clicked me');++x}

//   // wrapper.setProps({ disabled:  true, onLoginButtonPressed })
//   // const button = wrapper.find('View > View > View > TouchableOpacity').first()
//   // button.simulate('press')
//   // t.is(x, 0) // not really a good condition :(

//   // const button$ = wrapper.find('View > View > View > TouchableOpacity').first()
//   // wrapper.setProps({ disabled: false,  fetching: false, onLoginButtonPressed })
//   // button$.simulate('press')
//   // t.is(x, 1)
// })
