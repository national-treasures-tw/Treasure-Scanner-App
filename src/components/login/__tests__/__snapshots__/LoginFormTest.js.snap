// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component exists 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.8)",
      "borderRadius": 4,
      "flexDirection": "column",
      "paddingTop": 20,
      "width": 562.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
        },
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#d1caca",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 15,
          "margin": 5,
          "width": 15,
        }
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      editable={false}
      keyboardType="email-address"
      onChangeText={undefined}
      onSubmitEditing={[Function]}
      placeholder="Email Address"
      returnKeyType="next"
      style={
        Object {
          "color": "#CCCCCC",
          "flex": 1,
          "fontFamily": "Futura",
          "fontSize": 34,
          "height": 30,
        }
      }
      underlineColorAndroid="transparent"
      value="Daredevil"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginHorizontal": 20,
        },
        Object {
          "borderBottomWidth": 0.5,
          "borderColor": "#d1caca",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 15,
            "margin": 5,
            "width": 15,
          },
          Object {
            "width": 16,
          },
        ]
      }
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      editable={false}
      keyboardType="default"
      onChangeText={undefined}
      onSubmitEditing={undefined}
      placeholder="Password"
      returnKeyType="go"
      secureTextEntry={true}
      style={
        Object {
          "color": "#CCCCCC",
          "flex": 1,
          "fontFamily": "Futura",
          "fontSize": 34,
          "height": 30,
        }
      }
      underlineColorAndroid="transparent"
      value="The devil of Hell's Kitchen"
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 10,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#e78720",
          "fontFamily": "Futura",
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      Forgot my password
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#e78720",
          "borderRadius": 4,
          "margin": 5,
          "marginBottom": 0,
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 40,
          "paddingRight": 40,
          "paddingTop": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#292929",
            "fontFamily": "Futura",
            "fontSize": 34,
            "textAlign": "center",
          }
        }
      >
        LOG IN
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontFamily": "Futura",
            "fontSize": 34,
          }
        }
      >
        New User? 
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#e78720",
              "fontFamily": "Futura",
              "fontSize": 34,
            }
          }
        >
          Sign up
        </Text>
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "color": "#808080",
          "fontFamily": "Futura",
          "fontSize": 34,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Cancel
    </Text>
  </View>
</View>
`;
